src/index.ts

```diff
@@ -80,95 +37,95 @@ export function act(name: string, fn: (data: any) => Promise<any>) {
-          result = AssertionResult.Match;
-        } catch (e) {
-          /* tslint:disable:no-console */
-          console.log("     : " + JSON.stringify(value));
-          console.log("     : !=");
-          console.log("     : " + a.snapshot);
-          /* tslint:enable:no-console */
-          result = AssertionResult.Miss;
-        }
-      }
-
-      // Save assertion result & new snapshot value
-      FLOWS = FLOWS.set(f.id, f.with({
-        assertions: f.assertions.set(a.id, a.with({
-          result, value: JSON.stringify(value),
-        })),
-      }));
-    });
-  }));
+  const parentFlow = currentFlow;
+  parentFlow.fns.push(async (data: any) => {
+    log("CHECK: %s", name);
+    const value = await fn ? fn(data) : data;
+    parentFlow.assertions.push({name, snapshot: JSON.stringify(value)});
+  });
 }
 
-export async function run(flows: List<Flow>) {
-  // Clear flows of snapshots
-  FLOWS = FLOWS.map((f) => f.with({
-    assertions: f.assertions.map((a) => a.with({ snapshot: null })),
-  }));
-
-  // Merge registered flows with provided flows (w/ snapshots)
-  for (const f of flows) {
-    const fMatch = FLOWS.find((f2) => f2.name === f.name);
-    if (fMatch) {
-      let mergedAssertions = f.assertions;
-      for (const a of f.assertions) {
-        const aMatch = fMatch.assertions.find((a2) => a2.name === a.name);
-        if (aMatch) {
-          mergedAssertions = mergedAssertions.set(aMatch.id, aMatch.with({
-            result: a.result,
-            snapshot: a.snapshot,
-          }));
-        } else {
-          mergedAssertions = mergedAssertions.push(a.with({ id: mergedAssertions.size }));
-        }
-      }
-      FLOWS = FLOWS.set(f.id, f.with({assertions: mergedAssertions}));
-    } else {
-      FLOWS = FLOWS.push(f.with({ id: FLOWS.size }));
-    }
-  }
+export async function run() {
+  for (const f of FLOWS) {
+    // Reset assertions results
+    f.assertions = [];
+
+    log(" FLOW: %s", f.name);
 
-  // Run flows
-  for (const f of FLOWS.toSeq()) {
-    /* tslint:disable-next-line:no-console */
-    console.log(" FLOW: " + f.name);
-    const fns = FLOW_FNS.get(f.id);
+    // Start with `null` as current value and thread returned results through
+    // each subsequent `act` or `check` function.
     let result = null;
-    for (const fn of fns) {
+    for (const fn of f.fns) {
       result = await fn(result);
     }
   }
 
   return FLOWS;
 }
+
+const error4002 = {
+  cause: "The request's URI points to a resource which does not exist.",
+  code: 4002,
+  message: "Requested resource not found",
+};
+const error5000 = {
+  cause: "An unknown error occured while processing this request.",
+  code: 5000,
+  message: "Internal server error",
+};
+
+function renderJson(res: http.ServerResponse, statusCode: number, body: any) {
+  res.setHeader("Content-Type", "application/json");
+  res.statusCode = statusCode;
+  res.end(JSON.stringify(body));
+}
+
+export function start() {
```

Comment:
We want to *remove* the web server component of the worker! Can you remove `start`?

Reply:
Hmm, but I thought it would make things easier for people if they didn't have to implement a basic webserver themselves when they don't need to.

This `start` method is totally opt-in, the `run` method still exists and is exported.

Can you run me through the steps a user would have to go through to deploy a checklist sdk/worker to Heroku without start?

_(The "server" component of the worker also has been greatly simplified, it doesn't require any external dependencies anymore to avoid imposing that on users and simply returns the `run` results when receiving a `POST` request to `/`)_

Reply:
Ohhh okay. I misunderstood your intent in your previous commits then. Thanks for elaborating!

Yeah, I think an opt-in web service with no dependencies, as you've designed, is a good idea. I hate to flip-flop, but can you revert your latest commit now that I understand your intent and totally buy into it, and we'll leave `start` in the SDK? 😅

Reply:
Done :smiley_cat: